packages/workflow/src/features/events/handler.ts

```diff
@@ -243,7 +253,9 @@ export async function fhirWorkflowEventHandler(
     case Events.DEATH_MARK_REG:
     case Events.MARRIAGE_MARK_REG:
       response = await markEventAsWaitingValidationHandler(request, h, event)
```

Comment:
My gut feeling is that there would almost need to be some sort of generic and very robust

```
function rejectRecordForUnexpectedError(id: string) {}
```

function that could be called from anywhere when something unexpected happens. The function would ensure no record can ever be left in an incoherent state. Even the logic now in country config validate's catch has so many steps that I'm a little concerned what happens if only half of them are successful..